agent.go

```diff
@@ -200,6 +206,99 @@ func (c *container) getProcess(execID string) (*process, error) {
 	return proc, nil
 }
 
+// setSandboxStorage sets the sandbox level reference
+// counter for the sandbox storage.
+// This method also returns a boolean to let
+// callers know if the storage already existed or not.
+// It will return true if storage is new.
+//
+// It's assumed that caller is calling this method after
+// acquiring a lock on sandbox.
+func (s *sandbox) setSandboxStorage(path string) bool {
```

Comment:
It seems both `setSandboxStorage` and `unSetSandboxStorage` are called under the protection of the sandbox lock. The same lock is used to protect the storage map IIUC. Then there is no need for a sbs lock just to protect the refcount.

Reply:
There's also `GetRefCount()` being called by `unsetAndRemoveSandboxStorage()` but again, this is protected by a sandbox lock at some point.
I agree the sbs lock might be a bit too much here, but we need to be careful with this in case we rework the locking to a finer granularity in the future.